scripts/release.sh

```diff
@@ -0,0 +1,291 @@
+            record_url=$(sed -n '/url:/{s/url://g;s/^[[:space:]]*//g;s/"//g;p;}' <<< "$record_info")
+            record_version=$(sed -n '/from:/{s/from://g;s/^[[:space:]]*//g;s/"//g;p;}' <<< "$record_info")
+            
+            # figure out latest version (visit link?)
+            #dependency_latest_version=$(gh repo view $record_url --json latestRelease -q .latestRelease.tagName)
+
+            # compare versions
+            #if [ $record_version != $dependency_latest_version ]; then
+                echo
+                echo "Current version of $record_name: "$record_version
+                #echo "Latest version of $record_name: "$dependency_latest_version
+            #fi
+        fi
+    done <<< "$submodule_info"
+    echo
+    echo "Please update the dependencies in Package.swift, SmartDeviceLink.podspec, and SmartDeviceLink-iOS.podspec appropriately, then press enter..."
+    read user_input 
+fi
+
+# 4 update changelog
+# TODO - insert a template into the changelog that includes the version the users have selected above.
+#echo "A template for this release has been inserted into the changelog.  Please update it."
+echo 
+echo "Please update CHANGELOG.md, then return here and press enter..."
+read user_input
+# TODO - check modified info before and after so we can detect if the user failed to update the file.
+
+
+# 5 generate documentation
+prompt_user "Would you like to automatically generate documentation with Jazzy"
+if [[ $? == 1 ]]; then
+    # Check if Jazzy is already installed, and if not then install jazzy
+    if [ -z "$(mdfind -name 'Jazzy')" ]; then
+        echo "Jazzy is not installed, attempting to install (may require your password)..."
+        sudo gem install jazzy
+    fi
+
+    # this runs Jazzy to generate the documentation
+    echo "Running Jazzy to generate documentation..."
+    jazzy --clean --objc --framework-root SmartDeviceLink --sdk iphonesimulator --umbrella-header SmartDeviceLink/public/SmartDeviceLink.h --theme theme --output docs
+fi
+
+# 6 Ensure that the RPC_SPEC has released to the master branch and update the submodule to point to the new release tag (or to the HEAD of master, if no release of the RPC_SPEC is occurring).
+echo 
+echo "Please check if there is a new release of the RPC_SPEC on https://www.github.com/smartdevicelink/rpc_spec"
+echo "If there is, please update the rpc_spec submodule to point to the newest commit on the master branch. Press enter to continue..."
+read user_input
+#TODO - phase ? - can this be automated.  Check version here.  Check version at site.
+#record_url=https://www.github.com/smartdevicelink/rpc_spec
+#submodule_latest_version=$(gh repo view $record_url --json latestRelease -q .latestRelease.tagName)
+
+# git commands
+echo
+echo "$develop_branch has already been checked out for you."
+
+prompt_user "Would you like to automatically run the git commands for this release"
```

Comment:
```suggestion
prompt_user "Would you like to walk through the git commands for this release"
```

Automatic sounds like it won't prompt the user, but it will.